### DISCUSSION TITLE
Extending unsafeSVG

### DISCUSSION THREAD

**Discussion by tommillard** [lit/lit#3240] [Ideas]
I use lit in my day job to build internal prototypes, and I find unsafeSVG really useful. I don't have to worry about untrusted content, so it gives me a very convenient way to get svg assets into my markup inline, so that their properties can be styled in css (fill, width, height etc.).

One code convention we have here is that all css is applied to name-spaced classes, we never write styles against element names, but this causes friction with my approach to svg assets. The only way I can hit an svg inserted using unsafeSVG is to reference its element name. I've been wondering whether I can extend unsafeSVG to add a second argument, a class name, which would be inserted into the svg string returned by the function.

I'm really out of my depth here though, so I'm just hanging this out here in the hope that someone might be able to point me in the right direction. I'm wanting to be able to write something like `${unsafeSVG(genericTickSVGAsset, "bhp-Modal_Tick")}`.

So far this is what I've written, from sniffing round in dev tools console logs as I go, but I know it's very brittle and hacky...

```ts
import { unsafeSVG } from "lit/directives/unsafe-svg.js";

export function unsafeSVGWithClass(svgValue: string, className: string) {
    let svgAsset = unsafeSVG(svgValue);
    
    svgAsset.values[0] = amendSVGToInsertClassName(svgAsset.values[0]);
    return svgAsset;
}

function amendSVGToInsertClassName(svgString) {
    // do the manipulation to add thr class name here.
    return svgString;

}
```

**Comment by justinfagnani** [comment]
> I've been wondering whether I can extend unsafeSVG to add a second argument, a class name, which would be inserted into the svg string returned by the function.

Where do you want to class name inserted? `unsafeSVG` doesn't have to insert just one element, do you want it on all top-level elements?

I don't think your approach is bad necessarily. I would worry about performance and doing the string manipulation every time, so I might check the input value against the previous and only manipulate when necessary. This means making a directive instead of a simple function.

I wonder if you could just wrap the input SVG string with a `<g>` element with the right class name. Alternative, why doesn't the SVG already have the class on it? What do other consumers of this SVG do?
